### ISSUE TITLE
Switch tests to Jest (Orig: Package doesn't build)

### ISSUE THREAD

**Issue by xerocross** [mobxjs/mobx-angular#87]
Is this repo deprecated?  I cloned the repo. Just running npm install throws pages and pages of errors and warnings about deprecated dependencies.  Then I tried building anyway.  It seems to build, but with many warning messages.  I try to run the tests, and it encounters some kind of fatal error and bails.

But it looks like a lot of people still use this package, so I'm confused about the state of it.

**Comment by adamkleingit** [comment]
Hi,
It's not deprecated. There are no errors that prevent building or using, just warnings, and I think all of them are originated from older versions of Angular CLI / modules.
Need to check which ones we can completely remove, and which can we update.

But it doesn't prevent from building/using the lib

**Comment by adamkleingit** [comment]
About the tests - I'm trying to shift them to using Jest, as they have been too flaky on the CI

**Comment by xerocross** [comment]
I am a fan of Jest.  I'll be interested to see if you make any further changes.  When I tried to build it, I did get dist files at the end as you say, but since there were problems with the testing, I found no way to verify that the dist files were correct and functioning as expected.

I actually did get non-fatal errors during the install process.  I think it was for some package in your dev dependencies (or their dependencies, etc) that wasn't strictly necessary for the build procedure though.

If you'd like a fresh set of eyes on anything, I could help.  Actually I intend to investigate a little further anyway because I like what this package offers.

**Comment by adamkleingit** [comment]
@xerocross I'm glad to hear that! Any set of eyes and hands would be more than welcome.
BTW - is there anything in particular that you want to add / fix?

**Comment by kubk** [comment]
Hi, @adamkleingit I'd like to help with this task. We want to switch from Karma to Jest and make tests green, is it correct? Currently, Karma tests are failing for me locally.

**Comment by adamkleingit** [comment]
Hi, @kubk thanks for the help!
Yes, it's actually 2 different tasks, but hopefully moving to Jest will help solve the issues with the tests, and make them less flaky.

**Comment by kubk** [comment]
Hi @adamkleingit What do you think about running Prettier across entire project? Currently tsling is complaining about double quotes, that's why I replaced them with single quotes. Can make a PR. Thank you.

**Comment by adamkleingit** [comment]
Strange, now on master I get errors when building for some @types/X libraries.
@kubk do you also get that?

**Comment by kubk** [comment]
Yes, it is reproducible for me too. I'll take a look and reply with a result.

**Comment by kubk** [comment]
Hi @adamkleingit I managed to fix the issue by updating typescript from 2.7 to 2.9 and setting `skipLibCheck` to true in tsconfig.json
There are still some notices generated by rollup but the build files are successfully generated:
```
> rollup -c rollup.config.js dist/mobx-angular.js > dist/mobx-angular.umd.js

⚠️   The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
https://github.com/rollup/rollup/wiki/Troubleshooting#this-is-undefined
dist/directives/mobx-autorun-sync.directive.js (1:17)
1: var __extends = (this && this.__extends) || (function () {
                    ^
2:     var extendStatics = Object.setPrototypeOf ||
3:         ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||

```
I think we also need to include `npm run build` to CI pipeline to make sure each PR is buildable. Can make a PR. Thoughts? Thank you.

**Comment by adamkleingit** [comment]
Merged. @kubk Thanks!
